### PR TITLE
Use PublicApiAnalyzer with NuGet.VisualStudio.Internal.Contracts

### LINKED ISSUE #10111: NuGet.VisualStudio.Internal.Contracts should use PublicAPIAnalyzers
`NuGet.VisualStudio.Internal.Contracts` is not packed into a nupkg, and is entirely for NuGet's own internal use. However, it defines the over-the-wire contracts for Code Spaces (and Live Share?) scenarios. Therefore, backwards compatibility is still very important. Hence we should use PublicApiAnalyzers to minimise the risk that someone accidentally adds a breaking change.

### PR DESCRIPTION
**Pull request by zivkan**
## Bug

Fixes: https://github.com/NuGet/Home/issues/10111
Regression: No  
* Last working version:   
* How are we preventing it in future:   

## Fix

Details: Opt NuGet.VisualStudio.Internal.Contracts into using PublicApiAnalyzer.

## Testing/Validation

Tests Added: No  
Reason for not adding tests:  no code changes
Validation:

### CODE REVIEW

**Comment by zivkan** [comment]
FYI: @rrelyea @dtivel @sbanni 

I'll need to update my branch if anyone merges a PR into dev that modifies `NuGet.VisualStudio.Internal.Contracts` before I merge this one, and if I merge this in, then anyone else working on the project needs to rebase their branch. Otherwise, we can get a green PR build that will break the official build.

**Review by nkolev92 — APPROVED** [review]

File: tools-local/ship-public-apis/Program.cs
Comment (erdembayar) [inline tools-local/ship-public-apis/Program.cs:113]: Where does this log this data? Is it part of CI/CD or somewhere during build? I haven't seen this yet.

Comment (zivkan) [reply tools-local/ship-public-apis/Program.cs:113]: No, it's not run as part of the CI build. It's something we need to manually run when we ship a GA version: https://github.com/NuGet/NuGet.Client/blob/dev/docs/nuget-sdk.md#shipping-nuget

**Review by erdembayar — APPROVED** [review]

**Comment by nkolev92** [comment]
🔔 I imagine the tests need re-run :)

**Comment by zivkan** [comment]
Still trying to get feedback from @rrelyea @sbanni and/or @dtivel about whether or not we should do this. From what I've implied from other comments I've seen, by limiting the service to Codespaces, and excluding LiveShare, the RPC API compatibility might not be relevant, so maybe this isn't necessary.

**Pull request merged**
